hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/IsStillRunningService.java

```diff
@@ -0,0 +1,146 @@
+import com.hazelcast.spi.Operation;
+import com.hazelcast.spi.TraceableOperation;
+import com.hazelcast.spi.impl.NodeEngineImpl;
+import com.hazelcast.spi.impl.operationexecutor.OperationExecutor;
+import com.hazelcast.spi.impl.operationexecutor.OperationRunner;
+import com.hazelcast.spi.impl.operationservice.impl.operations.IsStillExecutingOperation;
+import com.hazelcast.spi.impl.operationservice.impl.operations.TraceableIsStillExecutingOperation;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Checks if an Operation is still running. This is useful when an operation isn't executed (e.g. got lost due to an
+ * OOME on the remote side) and no response is ever send.
+ *
+ * The behavior needs to be cleaned up.
+ *
+ * Currently we only detect if an operation is actually running, not if it is in a queue somewhere. So it can mean that you
+ * assume an operation isn't running, even though it is queued. Another issue is that the actual IsStillRunningOperation
+ * isn't executed quick enough and then the system still decides that the operation isn't running.
+ *
+ * Instead of letting every invocation ask the remote system if an operation is still running, let the remote system
+ * send a batch of all the call-id's for a particular member of the operations that are running.
+ */
+public class IsStillRunningService {
+
+    private static final int IS_EXECUTING_CALL_TIMEOUT = 5000;
+
+    private final ILogger logger;
+    //private final OperationServiceImpl operationService;
+    private final NodeEngineImpl nodeEngine;
+    private final OperationExecutor operationExecutor;
+
+    public IsStillRunningService(OperationExecutor operationExecutor, NodeEngineImpl nodeEngine, ILogger logger) {
+        this.operationExecutor = operationExecutor;
+        this.logger = logger;
+        this.nodeEngine = nodeEngine;
+    }
+
+    public boolean isOperationExecuting(Invocation invocation) {
+        // ask if op is still being executed?
+        Boolean executing = Boolean.FALSE;
+        try {
+            Operation isStillExecuting = createCheckOperation(invocation);
+
+            Invocation inv = new TargetInvocation(
+                    invocation.nodeEngine, invocation.serviceName, isStillExecuting,
+                    invocation.getTarget(), 0, 0, IS_EXECUTING_CALL_TIMEOUT, null, true);
+            Future f = inv.invoke();
+            invocation.logger.warning("Asking if operation execution has been started: " + toString());
+            executing = (Boolean) invocation.nodeEngine.toObject(f.get(IS_EXECUTING_CALL_TIMEOUT, TimeUnit.MILLISECONDS));
+        } catch (Exception e) {
+            invocation.logger.warning("While asking 'is-executing': " + toString(), e);
+        }
+        invocation.logger.warning("'is-executing': " + executing + " -> " + toString());
+        return executing;
+    }
+
+    private Operation createCheckOperation(Invocation invocation) {
```

Comment:
I prefer private methods after public ones. Don't we have a checkstyle rule enforcing this?